clang/runtime/dpct-rt/include/math.hpp

```diff
@@ -242,6 +242,27 @@ inline unsigned unordered_compare_mask(const sycl::marray<T, 2> a,
       .as<sycl::vec<unsigned, 1>>();
 }
 
+template <typename T>
+inline T bfe(T source, uint32_t bit_start, uint32_t num_bits) {
+  if (!num_bits)
+    return 0;
+  bit_start &= 0xFF;
+  num_bits &= 0xFF;
```

Comment:
I'm not sure I understand the reason to bitwise AND with the lowest 8 bits.  
Of course, if someone sends in a value greater than 255, it is somewhat nonsensical, but if we want to put in guardrails, wouldn't a branch with an error be better than masking these values?

Reply:
I try to use dpct::bfe as a mapping of cub::BFE and PTX bfe instruction. To keep the same semantic with the wording in https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#integer-arithmetic-instructions-bfe ,I masked `bit_start` and `num_bits`.

Reply:
I would be inclined to remove it.  I'm not sure if their code actually uses the masking, or if that is just used in the documentation for descriptive purposes, but the affect would be to accept very large values and treat them as their masked input versions.  

Any pre-migration code relying on this part of the semantics is almost surely relying on a bug.  I can't imagine someone sending in a bit start of 258 and expecting a bit start of 2 to be what is applied.

Additionally, this adds a couple extra (hopefully no-op) operations to every call.

Reply:
I updated this function so that it behaves the same as PTX in edge cases(Eg. bit_start/num_bits out of range, or `T` is signed integer). As far as I know, `std::min` can be optimized https://godbolt.org/z/rbjT7Wb9r , but I'm not sure yet if it will be optimized on GPU.

Reply:
I updated this function to keep behavior stable for boundary cases(Eg. bit_start/num_bits out of range, or `T` is signed integer). As far as I know, `std::min` can be optimized https://godbolt.org/z/rbjT7Wb9r , but I'm not sure yet if it will be optimized on GPU.